bitcoin/src/blockdata/script/owned.rs

```diff
@@ -194,6 +195,15 @@ impl ScriptBuf {
     /// This method doesn't (re)allocate.
     pub fn from_bytes(bytes: Vec<u8>) -> Self { ScriptBuf(bytes) }
 
+    /// Parse a Script in ASM format.
+    ///
+    /// NOTE: Receiving an [Ok] result from this function **ONLY** means that
+    /// our script interpreter could interpret the script, but it gives **no
+    /// guarantees** on the validity of the resulting script.
+    pub fn parse_asm(asm: &str) -> Result<Self, ParseAsmError> {
+        parse_asm(asm)
```

Comment:
Why not write the logic here directly?

Reply:
Idk, I mimicked what fmt_asm did. And also the helper methods are all in script/mod.rs.

Reply:
I think those are from the time before unsized `Script`. We could move them eventually.